cms/static/js/views/import.js

```diff
@@ -2,241 +2,319 @@
+         *
+         */
+        var initEventListeners = function () {
+            window.onbeforeunload = function () {
+                if (current.stage <= STAGE.UNPACKING) {
+                    return gettext('Your import is in progress; navigating away will abort it.');
+                }
+            }
+        };
 
         /**
-         * Manipulate the DOM to reflect current status of upload.
-         * @param {int} stageNo Current stage.
+         * Stores in a cookie the current import data
+         *
+         * @param {boolean} [completed=false] If the import has been completed or not
          */
-        var updateStage = function (stageNo){
-            var all = $('ol.status-progress').children();
-            var prevList = all.slice(0, stageNo);
-            _.map(prevList, function (elem){
-                $(elem).
-                    removeClass("is-not-started").
-                    removeClass("is-started").
-                    addClass("is-complete");
-                updateCog($(elem), false);
-            });
-            var curList = all.eq(stageNo);
-            curList.removeClass("is-not-started").addClass("is-started");
-            updateCog(curList, true);
+        var storeImport = function (completed) {
+            $.cookie(COOKIE_NAME, JSON.stringify({
+                file: file,
+                date: moment().valueOf(),
+                completed: completed || false
+            }));
         };
 
         /**
-         * Check for import status updates every `timeout` milliseconds, and update
-         * the page accordingly.
-         * @param {string} url Url to call for status updates.
-         * @param {int} timeout Number of milliseconds to wait in between ajax calls
-         *     for new updates.
-         * @param {int} stage Starting stage.
+         * Sets the Import on the "success" status
+         *
+         * If it wasn't already, marks the stored import as "completed",
+         * and updates its date timestamp
          */
-        var getStatus = function (url, timeout, stage) {
-            var currentStage = stage || 0;
-            if (currentStage > 1) { CourseImport.okayToNavigateAway = true; }
-            if (CourseImport.stopGetStatus) { return ;}
-
-            if (currentStage === 4) {
-                // Succeeded
-                CourseImport.displayFinishedImport();
-                $('.view-import .choose-file-button').html(gettext("Choose new file")).show();
-            } else if (currentStage < 0) {
-                // Failed
-                var errMsg = gettext("Error importing course");
-                var failedStage = Math.abs(currentStage);
-                CourseImport.stageError(failedStage, errMsg);
-                $('.view-import .choose-file-button').html(gettext("Choose new file")).show();
-            } else {
-                // In progress
-                updateStage(currentStage);
+        var success = function () {
```

Comment:
@AkA84 sorry for not bringing it earlier - I had subtle feeling that something is slightly wrong here but was unable to express it. Now I can: this success function probably should be a method on `CourseImport` object - it already have `start`, `error`, `resume` and `reset`, so having `success` on it feels natural. This would also allow to make `deferred` an internal member of `CourseImport`.